src/web/app/components/session-edit-form/session-edit-form.component.ts

```diff
@@ -134,6 +134,35 @@ export class SessionEditFormComponent {
     });
   }
 
+  /**
+   * Triggers the change of the model when the submission opening date changes.
+   */
+  triggerSubmissionOpeningDateModelChange(field: string, date: DateFormat): void {
+    const minDate: DateFormat = this.minDateForSubmissionStart;
+    if (JSON.stringify(minDate) === JSON.stringify(date)) {
+      const minTime: TimeFormat = this.minTimeForSubmissionStart;
+      this.configureSubmissionOpeningTime(minTime);
+      this.model.submissionStartTime = minTime;
+    }
+    this.modelChange.emit({
+      ...this.model,
+      [field]: date,
+    });
+  }
+
+  /**
+   * Configures the time for the submission opening time.
+   */
+  configureSubmissionOpeningTime(time : TimeFormat) : void {
+    if (time.hour < 24 && time.minute > 0) {
+      // Case where minutes is not 0 since the earliest time with 0 minutes is the hour before
+      time.hour += 1;
+      time.minute = 0;
+    } else if (time.hour === 23 && time.minute > 0) {
+      time.minute = 59;
+    }
```

Comment:
time.hour === 23 && time.minute > 0 falls under the if branch case as well, I believe that the branches should be switched, could we also add unit tests for this method as well? (especially given that it deals with time)